tests/merch.test.ts

```diff
@@ -0,0 +1,63 @@
+import { ForbiddenError } from 'routing-controllers';
+import { UserAccessType } from '../types';
+import { DatabaseConnection, MerchFactory, UserFactory, PortalState } from './data';
+import { ControllerFactory } from './controllers';
+
+beforeAll(async () => {
+  await DatabaseConnection.connect();
+});
+
+beforeEach(async () => {
+  await DatabaseConnection.clear();
+});
+
+afterAll(async () => {
+  await DatabaseConnection.clear();
+  await DatabaseConnection.close();
+});
+
+describe('collection editing', () => {
+  test('Set collection to archive', async () => {
+    const conn = await DatabaseConnection.get();
+    const itemOpt = MerchFactory.fakeOption();
+    const [item] = MerchFactory.itemsWith({
+      options: [itemOpt],
+    });
+    const [collection] = MerchFactory.collectionsWith({
+      items: [item],
+      archived: false,
+    });
+    item.collection = collection;
+    itemOpt.item = item;
```

Comment:
@shravanhariharan2 didn't occur to me earlier but maybe we should set `item.collection` and `option.item` in the `MerchFactory` methods.

Reply:
I'm not sure how necessary it would be since all the read operations in our repositories do joins on the parent. @michl1001 does the test case fail if the parents aren't set here?

Reply:
Those are nullable fields, so we need to set them.

Reply:
The test case doesn't fail without them but I don't think the code driectly accesses those values